crates/circuit/src/circuit_data.rs

```diff
@@ -995,6 +996,17 @@ impl CircuitData {
         self.param_table.clear();
     }
 
+    pub fn count_ops(&mut self) -> IndexMap<String, i32> {
+        let mut ops_count: IndexMap<String, i32> = IndexMap::new();
+        let circuit: &Vec<PackedInstruction> = &self.data;
+        for instruction in circuit {
+            *ops_count
+                .entry(instruction.op.view().name().to_string())
+                .or_insert(0) += 1;
+        }
+        ops_count
```

Comment:
The thing missing from the python implementation in this code is the final sort by counts, making the higher count instructions first. You can do that with:

```suggestion
        ops_count.par_sort_by(|_k1, v1, _k2, v2| v2.cmp(v1));
        ops_count
```

which will do a multithreaded sort inplace and put the larger counts first. You can drop the `par_` prefix to do it serially. It's probably worth benchmarking to see which is faster, because doing it in parallel doesn't always improve performance.

Reply:
I saw no significant difference in benchmarking between the parallel sort, sequential sort, or default hashing algorithm.